src/games/services/game-launcher.service.ts

```diff
@@ -28,55 +30,61 @@ export class GameLauncherService {
    * @memberof GameLauncherService
    */
   async launch(gameId: string): Promise<Game> {
-    let game = await this.gamesService.getById(gameId);
+    return await this.mutex.runExclusive(async () => {
```

Comment:
I'm not a async-mutex user, but is there a way we can instead aquire a lock instead of a nasty callback?

Reply:
We can aquire it, of course, but the callback is cleaner solution, since it doesn't require manual release.